src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.SolutionChecksumUpdator.cs

```diff
@@ -128,21 +133,40 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
 
             private async Task UpdateSolutionChecksumAsync(CancellationToken cancellationToken)
             {
-                // hold onto previous snapshot
-                var previousSnapshot = _lastSnapshot;
+                using (await _gate.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
+                {
+                    // hold onto previous snapshot
+                    var previousSnapshot = _lastSnapshot;
 
-                // create a new one (incrementally update the snapshot)
-                _lastSnapshot = await _checksumService.CreateChecksumAsync(_workspace.CurrentSolution, cancellationToken).ConfigureAwait(false);
+                    // create a new one (incrementally update the snapshot)
+                    _lastSnapshot = await _checksumService.CreateChecksumAsync(_service.Workspace.CurrentSolution, cancellationToken).ConfigureAwait(false);
```

Comment:
why are you creating hte checksum while in a lock?  Shouldn't you create the checksum out of the lock, and then only assign the checksum in the lock?